File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) 
    {
        int maxProfit = 0;
        int min = prices[0];
        
        for(int i =0;i <prices.length;i++)
        {
            if(prices[i]<min)
            {
                min = prices[i];
            }
            else
            {
                int profit = prices[i]-min;
                maxProfit = Math.max(maxProfit,profit);
            }
            
        }
        return maxProfit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_price = float('inf')
        max_profit = 0
        for i in prices:
            if(min_price>i):
                min_price = i;
            else:
                max_profit = max(max_profit, i-min_price)
                
        return max_profit
                
            
        
    
        
                
        
        